subxt/src/custom_values/custom_value_address.rs

```diff
@@ -19,3 +22,27 @@ impl CustomValueAddress for str {
         self
     }
 }
+
+/// A static address to a custom value.
+pub struct StaticAddress<R> {
+    name: &'static str,
+    phantom: PhantomData<R>,
+}
```

Comment:
The main thing to add here is the validation hash stuff (see how static constant addresses are defined). Then it can be checked in the `at` call (look at constants again for inspiration; it's pretty much the same idea :))